src/api/variables/mutate/updateVariable.ts

```diff
@@ -0,0 +1,70 @@
+import {
+  VarFormSchemaType,
+  VarUpdatedSchema,
+  VarUpdatedSchemaType,
+} from "../schema";
+import { useMutation, useQueryClient } from "@tanstack/react-query";
+
+import { apiFactory } from "~/api/utils";
+import { useApiKey } from "~/util/store/apiKey";
+import { useNamespace } from "~/util/store/namespace";
+import { useToast } from "~/design/Toast";
+import { varKeys } from "..";
+
+const updateVar = apiFactory({
+  url: ({ namespace, name }: { namespace: string; name: string }) =>
+    `/api/namespaces/${namespace}/vars/${name}`,
+  method: "PUT",
+  schema: VarUpdatedSchema,
+});
+
+export const useUpdateVar = ({
+  onSuccess,
+}: {
+  onSuccess?: (data: VarUpdatedSchemaType) => void;
+} = {}) => {
+  const apiKey = useApiKey();
+  const namespace = useNamespace();
+  const { toast } = useToast();
+  const queryClient = useQueryClient();
+
+  if (!namespace) {
+    throw new Error("namespace is undefined");
+  }
+
+  const mutationFn = ({ name, content, mimeType }: VarFormSchemaType) =>
+    updateVar({
+      apiKey: apiKey ?? undefined,
+      payload: content,
+      urlParams: {
+        namespace: namespace,
+        name,
+      },
+      headers: {
+        "content-type": mimeType,
+      },
+    });
+
+  return useMutation({
+    mutationFn,
+    onSuccess: (data, variables) => {
+      queryClient.invalidateQueries(
```

Comment:
Since this is used for creating and updating, we need to update the cache for the list as well as the single variable. Technically, only one operation is needed in each case, but for now, I don't think it's worth adding that kind of complexity to the code before we update the backend API.

Reply:
I agree! Maybe this would be worth a comment just to make sure we don't accidentally identify this as a bug in the future

Reply:
solved in 0e506ce7ff06b99a986808ab3541bbca66612e3d